notify/factory.go

```diff
@@ -0,0 +1,98 @@
+package notify
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/prometheus/alertmanager/notify"
+
+	"github.com/grafana/alerting/receivers"
+	"github.com/grafana/alerting/receivers/alertmanager"
+	"github.com/grafana/alerting/receivers/dinding"
+	"github.com/grafana/alerting/receivers/discord"
+	"github.com/grafana/alerting/receivers/email"
+	"github.com/grafana/alerting/receivers/googlechat"
+	"github.com/grafana/alerting/receivers/kafka"
+	"github.com/grafana/alerting/receivers/line"
+	"github.com/grafana/alerting/receivers/opsgenie"
+	"github.com/grafana/alerting/receivers/pagerduty"
+	"github.com/grafana/alerting/receivers/pushover"
+	"github.com/grafana/alerting/receivers/sensugo"
+	"github.com/grafana/alerting/receivers/slack"
+	"github.com/grafana/alerting/receivers/teams"
+	"github.com/grafana/alerting/receivers/telegram"
+	"github.com/grafana/alerting/receivers/threema"
+	"github.com/grafana/alerting/receivers/victorops"
+	"github.com/grafana/alerting/receivers/webex"
+	"github.com/grafana/alerting/receivers/webhook"
+	"github.com/grafana/alerting/receivers/wecom"
+)
+
+var receiverFactories = map[string]func(receivers.FactoryConfig) (NotificationChannel, error){
```

Comment:
`NotificationChannel` is a legacy Grafana concept, is it possible to get rid of it? I'm fine with this part of the separate PR as this PR is already kind of hard to review as it is right now.

Reply:
It's just a composition of two interfaces. I think we can but not in this PR. We need to move code that builds integrations from Grafana first.